src/__swaps__/screens/Swap/components/CoinRow2.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useMemo } from 'react';
+import { ButtonPressAnimation } from '@/components/animations';
+import { Box, HitSlop, Inline, Stack, Text } from '@/design-system';
+import { TextColor } from '@/design-system/color/palettes';
+import { CoinRowButton } from '@/__swaps__/screens/Swap/components/CoinRowButton';
+import { BalancePill } from '@/__swaps__/screens/Swap/components/BalancePill';
+import Animated from 'react-native-reanimated';
+import { StyleSheet } from 'react-native';
+import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
+import { UniqueId } from '@/__swaps__/types/assets';
+import { userAssetsStore } from '@/state/assets/userAssets';
+
+export const CoinRow2 = ({ assetId, output, onPress }: { assetId: string; output?: boolean; onPress: (assetId: UniqueId) => void }) => {
+  const { AnimatedSwapStyles } = useSwapContext();
+
+  const asset = userAssetsStore(state => state.getUserAsset(assetId));
```

Comment:
instead of subscribing to the whole asset here, we should create `Cell` components for each piece of data and only subscribe to that data inside of that Cell.